### PR TITLE
Clean up bokeh check

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref https://github.com/dask/dask/pull/2844 I should have done more digging on when that was added.

cc @bryevdv

### CODE REVIEW

File: dask/diagnostics/profile_visualize.py [hunk 1]
```diff
@@ -169,16 +168,9 @@ def visualize(profilers, file_path=None, show=True, save=True, **kwargs):
...
 
-    if LooseVersion(bokeh.__version__) >= "0.12.10":
-        from bokeh.io import state
-        in_notebook = state.curstate().notebook
-    else:
-        from bokeh.io import _state
-        in_notebook = _state._notebook
-
-    if not in_notebook:
+    if not state.curstate().notebook:
```
Comment (bryevdv) [inline dask/diagnostics/profile_visualize.py]: I think you mean to do:
```
from bokeh.io import curstate
```

and 
```
if not curstate().notebook:
```

Comment (bryevdv) [reply dask/diagnostics/profile_visualize.py]: Also just to be completely clear, what `curstate().notebook` means is that `bokeh.io.output_notebook()` has been called, nothing else.

Comment (TomAugspurger) [reply dask/diagnostics/profile_visualize.py]: Right, IIUC the intent of this code is to determine whether we should display the visualization in the notebook or in a new tab.

File: dask/diagnostics/profile_visualize.py [hunk 2]
```diff
@@ -169,16 +168,9 @@ def visualize(profilers, file_path=None, show=True, save=True, **kwargs):
     The completed bokeh plot object.
     """
     bp = import_required('bokeh.plotting', _BOKEH_MISSING_MSG)
-    import bokeh
+    from bokeh.io.state import curstate
```
Comment (jakirkham) [inline dask/diagnostics/profile_visualize.py]: This should be `from bokeh.io import curstate`, right?

Comment (bryevdv) [reply dask/diagnostics/profile_visualize.py]: probably, `bokeh.io.state` exists from a recent refactoring, but `bokeh.io` is the stable public interface. If the intent is to support as many older versions as possible `bokeh.io` is the way to go

Comment (TomAugspurger) [reply dask/diagnostics/profile_visualize.py]: Seems like `bokeh.io` doesn't import `curstate`:

```python
In [1]: from bokeh.io import curstate
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-30e0c3d1f7ce> in <module>()
----> 1 from bokeh.io import curstate

ImportError: cannot import name 'curstate'

In [2]: import bokeh

In [3]: bokeh.__version__
Out[3]: '0.12.10'
```

Should I open an issue on the Bokeh issue tracker?

Comment (bryevdv) [reply dask/diagnostics/profile_visualize.py]: Well it looks like I unintentionally made difficult for you, unfortunately. I couldn't imagine any use cases when a regular user should ever use `curstate` so I put it in the "dev" category, which means it's only in `bokeh.io.state` as of `0.12.10` and not in the "normal user" public API that is in `bokeh.io`. I guess you are stuck with a version check (or try except) after all. My apologies.

**Comment by jakirkham** [comment]
Does it make sense to close this out in light of the discussion above or is there some other change we would like to make?

**Comment by TomAugspurger** [comment]
I think I'll just close it.

**Pull request closed without merging**